api/tests/unit/domain/usecases/find-user-for-oidc-reconciliation_test.js

```diff
@@ -0,0 +1,246 @@
+const { expect, sinon, catchErr, domainBuilder } = require('../../../test-helper');
+const findUserForOidcReconciliation = require('../../../../lib/domain/usecases/find-user-for-oidc-reconciliation');
+const { AuthenticationKeyExpired, DifferentExternalIdentifierError } = require('../../../../lib/domain/errors');
+
+describe('Unit | UseCase | find-user-for-oidc-reconciliation', function () {
+  let authenticationMethodRepository, userRepository;
+  let pixAuthenticationService, authenticationSessionService, oidcAuthenticationService;
+
+  beforeEach(function () {
+    authenticationMethodRepository = { findOneByUserIdAndIdentityProvider: sinon.stub() };
+    userRepository = { updateLastLoggedAt: sinon.stub() };
+    pixAuthenticationService = { getUserByUsernameAndPassword: sinon.stub() };
+    authenticationSessionService = { getByKey: sinon.stub(), update: sinon.stub() };
+    oidcAuthenticationService = { createAccessToken: sinon.stub(), saveIdToken: sinon.stub() };
+  });
+
+  it('should found pix user', async function () {
+    // given
+    pixAuthenticationService.getUserByUsernameAndPassword.resolves({ id: 2 });
+    authenticationSessionService.getByKey.resolves({
+      sessionContent: { idToken: 'idToken' },
+      userInfo: { firstName: 'Anne' },
+    });
+
+    // when
+    await findUserForOidcReconciliation({
+      email: 'ane.trotro@example.net',
+      password: 'pix123',
+      identityProviderCode: 'oidc',
+      authenticationSessionService,
+      pixAuthenticationService,
+      authenticationMethodRepository,
+      userRepository,
+    });
+
+    // then
+    expect(pixAuthenticationService.getUserByUsernameAndPassword).to.be.calledOnceWith({
+      username: 'ane.trotro@example.net',
+      password: 'pix123',
+      userRepository,
+    });
+  });
+
+  it('should found a user oidc authentication method', async function () {
```

Comment:
```suggestion
  it('should find a user oidc authentication method', async function () {
```

**Chipotage** : séparer les tests en plusieurs juste pour un autre expect c'est un peu overkill et on pourrait juste faire un test avec tous les expect.

Reply:
Je suis ok de squash les deux premiers tests mais pas les autres. Les tests c'est aussi de la doc, dont séparer pour expliciter ce que fait le usecase me paraît pertinent, même si ça implique de la répétition :)